mycroft/skills/phillips_hue/__init__.py

```diff
@@ -0,0 +1,283 @@
+            if self.ip == '':
+                self.ip = _discover_bridge()
+            self.bridge = Bridge(self.ip, self.username)
+            self.username = self.bridge.username
+        except DeviceNotFoundException:
+            self.speak_dialog('bridge.not.found')
+            return False
+        except PhueRegistrationException:
+            self.speak_dialog('connect.to.bridge')
+            i = 0
+            while i < 30:
+                time.sleep(1)
+                try:
+                    self.bridge = Bridge(self.ip)
+                except PhueRegistrationException:
+                    continue
+                else:
+                    break
+            if not self.connected:
+                self.speak_dialog('failed.to.register')
+                return False
+            else:
+                self.speak_dialog('successfully.registered')
+        self.all_lights = Group(self.bridge, 0)
+        return True
+
+    def initialize(self):
+        self.load_data_files(dirname(__file__))
+
+        turn_off_intent = IntentBuilder("TurnOffIntent")\
+            .require("TurnOffKeyword").build()
+        self.register_intent(turn_off_intent, self.handle_intent)
+
+        turn_on_intent = IntentBuilder("TurnOnIntent")\
+            .require("TurnOnKeyword").build()
+        self.register_intent(turn_on_intent, self.handle_intent)
+
+        activate_scene_intent = IntentBuilder("ActivateSceneIntent")\
+            .require("ActivateSceneKeyword").build()
+        self.register_intent(activate_scene_intent,
+                             self.handle_intent)
+
+        decrease_brightness_intent = IntentBuilder("DecreaseBrightnessIntent")\
+            .require("DecreaseBrightnessKeyword").build()
+        self.register_intent(decrease_brightness_intent,
+                             self.handle_intent)
+
+        increase_brightness_intent = IntentBuilder("IncreaseBrightnessIntent")\
+            .require("IncreaseBrightnessKeyword").build()
+        self.register_intent(increase_brightness_intent,
+                             self.handle_intent)
+
+        decrease_color_temperature_intent =\
+            IntentBuilder("DecreaseColorTemperatureIntent")\
+            .require("DecreaseColorTemperatureKeyword").build()
+        self.register_intent(decrease_color_temperature_intent,
+                             self.handle_intent)
+
+        increase_color_temperature_intent =\
+            IntentBuilder("IncreaseColorTemperatureIntent")\
+            .require("IncreaseColorTemperatureKeyword").build()
+        self.register_intent(increase_color_temperature_intent,
+                             self.handle_intent)
+
+        connect_lights_intent = \
+            IntentBuilder("ConnectLightsIntent") \
+            .require("ConnectLightsKeyword").build()
+        self.register_intent(connect_lights_intent,
+                             self.handle_intent)
+
+    def handle_intent(self, message):
```

Comment:
I'm very open to changing this (not a fan of the giant if block), but I'm not sure how else to structure things so that I'm not repeating the check at the start (confirming that we've established the initial connection to the bridge), and the error handling at the end.

Do you have any suggestions?